src/ctapipe/io/interpolating.py

```diff
@@ -0,0 +1,253 @@
+        if self.bounds_error:
+            self.interp_options["bounds_error"] = True
+        elif self.extrapolate:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = "extrapolate"
+        else:
+            self.interp_options["bounds_error"] = False
+            self.interp_options["fill_value"] = np.nan
+
+        self._interpolators = {}
+        self._secondary_interpolators = {}
+
+    @abstractmethod
+    def add_table(self, tel_id, input_table):
+        pass
+
+    def __call__(self, tel_id, time):
+        if tel_id not in self._interpolators:
+            if self.h5file is not None:
+                self._read_parameter_table(tel_id)  # might need to be removed
+            else:
+                raise KeyError(f"No table available for tel_id {tel_id}")
+
+    @abstractmethod
+    def _read_parameter_table(self, tel_id):
+        pass  # this method is called when you try to interpolate, but no table has been added.
+        # It will try adding a table from the hdf5 file specified in __init__
+
+
+class PointingInterpolator(Interpolator):
+    def __call__(self, tel_id, time):
+        """
+        Interpolate alt/az for given time and tel_id.
+
+        Parameters
+        ----------
+        tel_id : int
+            telescope id
+        time : astropy.time.Time
+            time for which to interpolate the pointing
+
+        Returns
+        -------
+        altitude : astropy.units.Quantity[deg]
+            interpolated altitude angle
+        azimuth : astropy.units.Quantity[deg]
+            interpolated azimuth angle
+        """
+
+        super().__call__(tel_id, time)
+
+        mjd = time.tai.mjd
+        az = u.Quantity(self._interpolators[tel_id](mjd), u.rad, copy=False)
+        alt = u.Quantity(self._secondary_interpolators[tel_id](mjd), u.rad, copy=False)
+        return alt, az
+
+    def add_table(self, tel_id, input_table):
+        """
+        Add a table to this interpolator
+
+        Parameters
+        ----------
+        tel_id : int
+            Telescope id
+        input_table : astropy.table.Table
+            Table of pointing values, expected columns
+            are ``time`` as ``Time`` column, ``azimuth`` and ``altitude``
+            as quantity columns for pointing and pointing correction data.
+        """
+
+        missing = {"time", "azimuth", "altitude"} - set(input_table.colnames)
+        if len(missing) > 0:
+            raise ValueError(f"Table is missing required column(s): {missing}")
+        for col in ("azimuth", "altitude"):
+            unit = input_table[col].unit
+            if unit is None or not u.rad.is_equivalent(unit):
+                raise ValueError(f"{col} must have units compatible with 'rad'")
+
+        if not isinstance(input_table["time"], Time):
+            raise TypeError("'time' column of pointing table must be astropy.time.Time")
+
+        input_table.sort("time")
```

Comment:
This is a side effect that I guess was there before this PR, but should be documented:  the input table gets sorted in-place when this function is called, which may be unexpected.  I guess at least we should add a note in the docstring that this function modifies the inputs.

Reply:
Probably best just add a `.copy()` so that we actually own the data and don't mutate someone else's table

Reply:
Maybe some memory overhead with that though... the  input tables could be large.  I guess there is no way make a "sorted view" of the table.